src/plugins/ResizeReorder/ResizeReorderCell.js

```diff
@@ -0,0 +1,258 @@
+import PropTypes from 'prop-types';
+import ResizerKnob from 'ResizerKnob';
+import ReorderHandle from 'ReorderHandle';
+import joinClasses from 'joinClasses';
+import cx from 'cx';
+
+/**
+ * A plugin that can make use of ResizerKnob and ReorderHandle to provide the
+ * resize and reorder functionality to the columns.
+ *
+ * Pass this to the `header` prop of the `Column` to invoke resizer/reorder handle.
+ * `onColumnResizeEndCallback` and `onColumnReorderEndCallback` activate the respective
+ * functionality. Either one or both can be used together.
+ *
+ */
+class ResizeReorderCell extends React.PureComponent {
+
+  static propTypes = {
+
+    /**
+     * Outer height of the cell.
+     */
+    height: PropTypes.number,
+
+    /**
+     * Outer width of the cell.
+     */
+    width: PropTypes.number,
+
+    /**
+     * Optional prop that if specified on the `Column` will be passed to the
+     * cell. It can be used to uniquely identify which column is the cell is in.
+     */
+    columnKey: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+    ]),
+
+    /**
+     * Optional prop that represents the rows index in the table.
+     * For the 'cell' prop of a Column, this parameter will exist for any
+     * cell in a row with a positive index.
+     *
+     * Below that entry point the user is welcome to consume or
+     * pass the prop through at their discretion.
+     */
+    rowIndex: PropTypes.number,
+
+    /**
+     * The height of the table.
+     */
+    tableHeight: PropTypes.number,
+
+    /**
+     * The left offset in pixels of the cell.
+     * Space between cell's left edge and left edge of table
+     */
+    left: PropTypes.number,
+
+    /**
+     * The left offset in pixels of the cell group.
+     */
+    cellGroupLeft: PropTypes.number,
+
+    /**
+     * Whether touch is enabled or not.
+     */
+    touchEnabled: PropTypes.bool,
+
+    /**
+     * If the component should render for RTL direction
+     */
+    isRTL: PropTypes.bool,
+
+    /**
+     * availableScrollWidth returned from ColumnWidths.
+     */
+    availableScrollWidth: PropTypes.number,
+
+    /**
+     * Maximum horizontal scroll possible.
+     */
+    maxScrollX: PropTypes.number,
+
+    /**
+     * Function to change the scroll position by interacting
+     * with the store.
+     */
+    scrollToX: PropTypes.func,
+
+    /**
+     * Whether the cells belongs to the fixed group
+     */
+    isFixed: PropTypes.bool,
+
+    /**
+     * Object consisting of keys and widths of the columns
+     * in the current cell group.
+     */
+    cellGroupColumnWidths: PropTypes.object,
+
+    /**
+     * The minimum width of the column.
+     */
+    minWidth: PropTypes.number,
+
+    /**
+     * The maximum width of the column.
+     */
+    maxWidth: PropTypes.number,
+  };
+
+  initialState = {
+    isColumnReordering: false,
+    displacement: 0,
+  };
+
+  state = { ...this.initialState };
+
+  updateReorderingData = (obj) => {
```

Comment:
I'm not really fond on how `ReorderHandle` is directly tied to `ResizeReorderCell`.
Instead of `updateReorderingData` which needs to know about the state, can we have something like `onReorder` and `onReorderEnd` callbacks passed to `ReorderHandle` that simply takes in the displacement?

Reply:
It is called from requestAnimationFrame as well to update the displacement. Then Need to third prop as well ?